Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM python:3.12
+
+RUN apt-get update && apt-get install -y \
+    redis-server \
```

Comment:
Does this need to be here if we are running redis using the official container already?

Reply:
From what I figured out during my testing, using the docker-compose.yml to build the redis server was inefficient, so I moved everything to the Dockerfile so that I could keep the docker-compose.yml file

Reply:
Switched it to the Docker Container Registry.